src/components/SamlConfiguration/index.jsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Collapsible, Icon } from '@edx/paragon';
+import SamlProviderConfigForm from './SamlProviderConfigForm';
+import SamlProviderDataForm from './SamlProviderDataForm';
+import { camelCaseObject, snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import LoadingMessage from '../LoadingMessage';
+import ErrorPage from '../ErrorPage';
+
+class SamlConfiguration extends React.Component {
+  state = {
+    providerConfig: undefined,
+    providerData: undefined,
+    error: undefined,
+    loading: true,
+    deleteEnabled: false,
+  };
+
+  componentDidMount() {
+    LmsApiService.getProviderConfig(this.props.enterpriseId)
+      .then((response) => {
+        this.setState({
+          providerConfig: response.data.results[0],
+          loading: false,
+        });
+      })
+      .catch(error => this.setState({
+        error,
+      }));
+    LmsApiService.getProviderData(this.props.enterpriseId)
+      .then((response) => {
+        this.setState({
+          providerData: response.data.results[0],
+          loading: false,
+        });
+      })
+      .catch(error => this.setState({
+        error,
+        loading: false,
+      }));
+  }
+  /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createProviderConfig = async (formData) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('name', this.props.enterpriseName);
+    transformedData.append('slug', this.props.enterpriseSlug);
+    transformedData.append('enterprise_customer_uuid', this.props.enterpriseId);
+    try {
+      const response = await LmsApiService.postNewProviderConfig(transformedData);
+      this.setState({ providerConfig: response.data });
```

Comment:
does this need to be 
```
this.setState({ ...this.state, providerConfig: response.data})
```

to preserve the rest of the state?

Reply:
setState on a specific value doesn't change the state of the other items. You may be thinking of nested state items.
Ie something like:
```
this.state = {
toplevel: {subkey: subvalue, subkey2: subvalue2}
}
```
in which case changing one of those keys would look like:
`this.setState(prevState => ({...prevState.toplevel, subkey2: newvalue }))`

Reply:
Another thing here is that I don't want to preserve any portion of the existing provider config. The response data gets me everything that's saved to that object in edx-platform, so I'd much rather get everything back than hope something got updated properly.

Reply:
yes you are correct, I misread the setState calls as if it was setStage(singleValue), but you are passing the object form so we are all set